k9mail-library/src/main/java/com/fsck/k9/mail/Flag.java

```diff
@@ -3,15 +3,37 @@
 
 /**
  * Flags that can be applied to Messages.
+ *
+ * Update: to support IMAP keywords ( custom flags ) this enum became
+ * a class. This class was constructed to resemble enums. Since Java
+ * enums are objects internally anyway ( implemented similar to these )
+ * this will not be noticably slower.
+ *
+ * The extra field bCustom denotes custom flags. These get an prefix
+ * attached internally. When using the flags with external servers,..
+ * one should use the realName() method.
  */
-public enum Flag {
-    DELETED,
-    SEEN,
-    ANSWERED,
-    FLAGGED,
-    DRAFT,
-    RECENT,
-    FORWARDED,
+public class Flag {
```

Comment:
i think this class could actually stay an enum, it's just a value object w/o any behaviour